src/components/Table/Table.tsx

```diff
@@ -452,42 +451,92 @@ class Table extends React.Component<Props, State> {
+      if (childData !== undefined && childData.length > 0) {
+        for (const key in childData) {
+          if (childData[key].component.props.data.length > 0) {
+            for (const index in childData[key].component.props.data) {
+              allChildData.push(childData[key].component.props.data[index]);
+            }
+          }
+        }
+        allRowId = allChildData.map((item: any) => {
+          return item.id;
+        });
+      }
+    }
 
-    if (!checkedStatus) {
-      selectedRows.splice(selectedRows.indexOf(uniqueId), 1);
-      this.setState({ selectedRows, allRowChecked: false }, () => {
-        this.rowSelectionCallback();
-      });
-    } else {
-      this.setState({ selectedRows: this.state.selectedRows.concat([uniqueId]) }, () => {
+    if (this.state.selectedRows.length > 0 && checkedStatus) {
+      for (const key in this.state.selectedRows) {
+        allRowId.push(this.state.selectedRows[key]);
+      }
+    }
+    allRowId.push(uniqueId);
+
+    this.setState({ selectedRows: allRowId }, () => {
+      if (!checkedStatus) {
+        if (allRowId.length > 1) {
+          for (let key = 0; key < allRowId.length; key++) {
+            if (selectedRows.indexOf(allRowId[key]) > -1) {
+              selectedRows.splice(selectedRows.indexOf(allRowId[key]), 1);
+            }
+          }
+        } else {
+          selectedRows.splice(selectedRows.indexOf(uniqueId), 1);
+        }
+        this.setState({ selectedRows, allRowChecked: false }, () => {
+          this.rowSelectionCallback();
+        });
+      } else {
         this.rowSelectionCallback();
         if (singleSelectRowCallback) {
           if (multipleCallBackValue && multipleCallBackValue.length) {
             const returnVal: any = {};
-
             multipleCallBackValue.forEach((item) => {
               returnVal[item] = rowData[item] !== undefined ? rowData[item] : '';
             });
-
             singleSelectRowCallback(returnVal);
           } else {
             singleSelectRowCallback(uniqueId);
           }
         }
-      });
-    }
+      }
+    });
   }
 
   // Function to select all the rows on click of header  checkbox
   toggleAllRowSelection = (checkedStatus: boolean) => {
-    const allRowId = this.state.data.map((item: any) => {
+
+    this.setState({ expandedRow: [] });
+
+    const allChildData: any = [];
+    if (this.props.nestedChildData !== undefined && this.props.nestedChildData.length > 0) {
+      for (const key in this.props.nestedChildData) {
+        if (this.props.nestedChildData[key].component.props.data.length > 0) {
+          for (const index in this.props.nestedChildData[key].component.props.data) {
+            allChildData.push(this.props.nestedChildData[key].component.props.data[index]);
+          }
+        }
+      }
+    }
+
+    allChildData.push(...this.state.data);
+    const allData = [...new Map(allChildData.map((o: any) => [o.id, o])).values()];
```

Comment:
@HiirenP don't use short forms like this, use childData as param or some valid name instead of just `o`